components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt

```diff
@@ -85,6 +91,66 @@ class QrFragmentTest {
         verify(qrFragment).tryOpenCamera(anyInt(), anyInt(), anyBoolean())
     }
 
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.N])
+    fun `WHEN running a device lower than P THEN startExecutorService should not called executed`() {
+        val qrFragment = spy(QrFragment.newInstance(mock()))
+
+        qrFragment.textureView = mock()
+        qrFragment.cameraErrorView = mock()
+        qrFragment.customViewFinder = mock()
+        whenever(qrFragment.textureView.isAvailable).thenReturn(true)
+        doNothing().`when`(qrFragment).startBackgroundThread()
+        doNothing().`when`(qrFragment).tryOpenCamera(anyInt(), anyInt(), anyBoolean())
+
+        qrFragment.onResume()
+
+        verify(qrFragment, never()).startExecutorService()
+    }
+
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.N])
+    fun `WHEN calling createCaptureSessionCompat on a device lower than P THEN use older API`() {
+        val qrFragment = spy(QrFragment.newInstance(mock()))
+        val camera = mock<CameraDevice>()
+        val imageSurface = mock<Surface>()
+        val surface = mock<Surface>()
+        val stateCallback = mock<CameraCaptureSession.StateCallback>()
+
+        qrFragment.textureView = mock()
+        qrFragment.cameraErrorView = mock()
+        qrFragment.customViewFinder = mock()
+        whenever(qrFragment.textureView.isAvailable).thenReturn(true)
+
+        qrFragment.createCaptureSessionCompat(camera, imageSurface, surface, stateCallback)
+
+        @Suppress("DEPRECATION")
+        verify(camera).createCaptureSession(listOf(imageSurface, surface), stateCallback, null)
+    }
+
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.P])
+    fun `WHEN calling createCaptureSessionCompat on a device higher than P THEN use newer api`() {
```

Comment:
@levimatheri I think the issue you were seeing on https://github.com/mozilla-mobile/android-components/pull/9491#discussion_r571672186 was related to the sdk version on which the test was running. Running with an sdk version lower than P the function [will not exist.](https://developer.android.com/reference/android/hardware/camera2/CameraDevice#createCaptureSession(android.hardware.camera2.params.SessionConfiguration))